### ISSUE TITLE
Trakt List(s) Import Manager does not seem to do anything

### ISSUE THREAD

**Issue by fr1sty** [umbrellaplug/umbrellaplug.github.io#123]
I tried to import my movies and tv shows from Trakt into my library, first time it worked but i cancelled it after few seconds because it seemed to have probems with scraping tv shows. Now i try to initiate it again, but it now only says updating library from trakt lists, and then completed.  But there are no movies added to my library. I rebooted, and reauthorized trakt, but made no difference.

**Comment by umbrellaplug** [comment]
On the import manager, on the list you're wanting to import. Hold ok to get the context menu and hit clear list database. Then try your import again. There's code to prevent you from adding the same items more than once that will prevent it from re-downloading without clearing that list from the database first.

**Comment by fr1sty** [comment]
Yes found the option! Tried it with my tv shows first, only 16 shows. Stopped importing after 4 shows. Then i repeated the steps again and it imported all my shows. Then i tried to import my movies, over 200 movies, nothing is happening even after clearing list database.

**Comment by umbrellaplug** [comment]
Yeah, strange. Would need some logs to see what's happening.

**Comment by fr1sty** [comment]
2022-11-16 16:57:52.447 T:9468     INFO <general>: CPythonInvoker(22, /storage/.kodi/addons/plugin.video.umbrella/umbrella.py): script successfully run
2022-11-16 16:57:52.772 T:9468     INFO <general>: Python interpreter stopped

from kodi.log is there any other log file with more information?

**Comment by umbrellaplug** [comment]
https://github.com/umbrellaplug/umbrellaplug.github.io/wiki#logs

Please follow the instructions and pictures here for setting up debugging on logs. The two lines you provided don't give me enough information to really do anything with. I'm happy to help, but I need information to assist.

**Comment by fr1sty** [comment]
I'm getting lines in my umbrella.log when choosing a video to play. But nothing when i'm importing Trakt lists.

**Comment by umbrellaplug** [comment]
Couple of basic setup questions. 

Do you have a folder set for your Movies/ TV Shows in library in settings of Umbrella? 

Can you give me an example of one of the lists you're trying to import so I can attempt to duplicate it here?

**Comment by fr1sty** [comment]
Yes I've set a folder, and there are a few movies in it which i added manually. 
Movies list https://trakt.tv/users/fr0sty/watchlist?display=movie&sort=rank,asc
Shows list https://trakt.tv/users/fr0sty/watchlist?display=show&sort=rank,asc (still sometimes it happens that the import does nothing, then i repeat it a few times and then it will loop the list, saying the items already exist) 
Does your code pull the items from the public pages? Our does it use the API to pull them?

**Comment by umbrellaplug** [comment]
Addon pulls from the API.

**Comment by umbrellaplug** [comment]
Do you have another list different than from your watchlists? I wouldn't be able to import those on my side. I can like any list you have it will show in my liked lists and I can attempt to import it, but unfortunately I cannot like you watchlist. :)

**Comment by fr1sty** [comment]
Ok, I hope I understand you right, i made an extra list and just added some movies. Don't see the list to import it yet, but probably some caching. 
https://trakt.tv/users/fr0sty/lists/mylist?sort=rank,asc

**Comment by umbrellaplug** [comment]
okay, i attempt to duplicate one of your watchlist as mine.

**Comment by umbrellaplug** [comment]
Can you try importing your watchlist again, but doing tv, then doing movie. and let me know if either fails or both.

**Comment by umbrellaplug** [comment]
I was able to bring in your trakt movie watchlist as a new list. I then attempted to duplicate your issue by importing the list. It worked. I looked over what you had described as your issue and tried it again and it appears you found a rare issue with clearing the watchlist from database. You can get around this by removing library.db from your kodi/userdata/addons/plugin.video.umbrella/ folder. I will push a correction for it in the next update. All other lists should work as expected for importing.

**Comment by fr1sty** [comment]
Thanks for all the effort. Tried again in different ways. Clearing, rebooting, importing again, without luck. And also removing the file you said on my raspberry its here /storage/.kodi/userdata/addon_data/plugin.video.umbrella/library.db. But it doesn't make any difference. Only the tv shows get imported, the movie list it just says its finished, even though some times i have to reboot before that will work. But the movie list never seems to work.

**Comment by umbrellaplug** [comment]
Yeah, it's a bug in the code. I'm getting pushing a version to the test repo now that I believe should resolve everything for you.

Let me know if you want to try it out and I'll send you the information for loading the test repo.

**Comment by umbrellaplug** [comment]
The test repo is loaded just like the original repo. 
Add Source > https://umbrellaplug.github.io/umbrellatest
You can then add the test repository from there and should be a version 6.5.86 available now. You will need to hold on your list on the import window and clear list database, but then your import should work. Please let me know your results.

**Comment by fr1sty** [comment]
After following your instructions my trakt list is in Kodi now! 

But there was a little difference in the total amount of movies. 
My watchlist on trakt says 303, but in umbrella it says 286 even after adding new movies 

I was wondering if new movies would be added automatically now, but nothing seemed to happen. 

Then I looked around in the settings under Trakt and found 'Sync on Kodi Start' under Auto Trakt collections syncing. I enabled this and rebooted Kodi. 
And now its adding everything Trakt has ever scrobbled, does this mean to happen? :) 

Still busy at the moment so don't know if the latest added movie will be in my list. 
Was just wondering if this was the right way to getting the watchlists in sync with umbrella.

**Comment by umbrellaplug** [comment]
There's also settings in your library settings of the addon for the library service. It runs by default every 6 hours and will add any new changes detected. If you have a list that isn't updating correctly you can use the context menu to clear the list database when you're on the import list manager window. You can use the slider in the settings to change the interval of how often it updates your library as well. Trakt is resynced every 15 minutes by default and library every 6 hours by default when enabled. Otherwise you can always just click "import now" at the list import window to update list right then.

**Comment by fr1sty** [comment]
I understand that the Kodi library also needs to update, but it seems that the umbrella plugin does not see the changes to my Trakt watchlist. If I look at the List(s) Import Manager it displays 286 movies, even though i had 303. I even removed a lot of movies now ,so i have less than 286 but still the import manager keeps the same amount not doing anything anymore when updating.

**Comment by umbrellaplug** [comment]
The list manager window should update with the correct numbers every 15 minutes or however often you have trakt set to sync. Keep in mind you've got multiple things going here. Trakt Sync to get your lists, import to import new items, and library update which just scans all imported items for changes that need to be updated. For most people it's pretty sufficient to have their lists update daily though honestly. I have been seeing some weird behaviors with the watchlist import. The first time I imported I had a movie list as well and items were missing. I cleared the database for it and re-imported and those items came in. I'll be working to fine tune it a little more.

**Comment by fr1sty** [comment]
The Trakt service sync interval is on 15 minutes. But it seems that only the Force Trakt Sync to local database option makes the numbers correct in the List(s) Import Manager and does import the shows I added earlier on.

**Comment by fr1sty** [comment]
Yes the scrobble is enabled. The day before yesterday I removed the extra list (which i made for testing purposes). It only disappeared just now from the list manager after the force trakt sync. I run kodi on libreelec on raspberry pi, and it's always on. So it seems like the cronjob for syncing every 15 minutes is not doing anything, at least not the same thing as the force sync option.

**Comment by umbrellaplug** [comment]
Not sure, I have a pi but don't run any library stuff on it due to the size of my card. I can check my trakt lists on it though and see. Logs are always helpful in scenario's like that. I'd say at least reboot it once to make sure the service is fired off.

**Comment by fr1sty** [comment]
No syncing yet, maybe this helps?
`[2022-11-22 19:28:36] [ Umbrella: INFO ]: #################### STARTING TRAKT SCHEDULING ################
[2022-11-22 19:28:36] [ Umbrella: INFO ]: #################### SCHEDULED TIME FRAME 1 HOURS ###############
[2022-11-22 20:28:39] [ Umbrella: DEBUG ]: From func name: resources.lib.menus.tvshows.super_info() Line # :1105
                       msg : tvshowtitle: (Endeavour: Everest) missing tmdb_id: ids={imdb: , tmdb: , tvdb: 276632}
[2022-11-22 20:29:06] [ Umbrella: ERROR ]: From func name: /resources/lib/modules/library.py.silent() Line # :909
                       msg : TypeError -> '>' not supported between instances of 'NoneType' and 'int'
`

**Comment by umbrellaplug** [comment]
One hour is the library sync service. That's an interesting one. Show is missing ids. I'll see if I can duplicate it here.

**Comment by umbrellaplug** [comment]
Can you try the latest version on the test repo. 6.5.90
It has additional logging for library import. I'd like to see a full log of what you're getting on the initial import of your collection when kodi starts up please.

**Comment by fr1sty** [comment]
Got some results, only after an hour, nothing after 15 minutes.

`[2022-11-23 11:37:45] [ Umbrella: INFO ]: ########   CURRENT Umbrella VERSIONS REPORT   ########
[2022-11-23 11:37:45] [ Umbrella: INFO ]: ##   Platform: linux
[2022-11-23 11:37:45] [ Umbrella: INFO ]: ##   Kodi Version: 19.3 (19.3.0) Git:19.3-Matrix
[2022-11-23 11:37:45] [ Umbrella: INFO ]: ##   python Version: 3.8.9
[2022-11-23 11:37:45] [ Umbrella: INFO ]: ##   plugin.video.umbrella Version: 6.5.90
[2022-11-23 11:37:45] [ Umbrella: INFO ]: ##   repository.umbrella Version: 2.1.42
[2022-11-23 11:37:45] [ Umbrella: INFO ]: ##   script.module.cocoscrapers Version: 1.0.5
[2022-11-23 11:37:45] [ Umbrella: INFO ]: ######   UMBRELLA SERVICE ENTERING KEEP ALIVE   #####
[2022-11-23 11:37:49] [ Umbrella: INFO ]: #################### STARTING TRAKT SCHEDULING ################
[2022-11-23 11:37:49] [ Umbrella: INFO ]: #################### SCHEDULED TIME FRAME 1 HOURS ###############
[2022-11-23 12:37:51] [ Umbrella: DEBUG ]: From func name: resources.lib.menus.tvshows.super_info() Line # :1105
                       msg : tvshowtitle: (Endeavour: Everest) missing tmdb_id: ids={imdb: , tmdb: , tvdb: 276632}
[2022-11-23 12:38:19] [ Umbrella: DEBUG ]: From func name: resources.lib.modules.library.silent() Line # :913
                       msg : tvshowtitle: (Outlander) ids={imdb: tt3006802, tmdb: 56570, tvdb: 270408} was not added due to exception: '>' not supported between instances of 'NoneType' and 'int':
`

**Comment by umbrellaplug** [comment]
These are errors being logged as expected. If there's an issue with a show it's going to throw it into the log and not import it and move into the next show.

15 minutes is how often trakt syncs. 

Library service is set in library via a slider it's default is every 6 hours. 

Trakt collection sync (which is what you're using) has its own service for importing that runs based on the slider in the trakt tab of settings. Default is every hour I believe. 

It seems we're getting off track here. You originally opened an issue for lists not importing from the import manager window. Is that working now? 

You then mentioned your trakt list not syncing. 

Now you're asking about trakt collection syncing. 

Can you provide some clarity of what your current issue is at this point and steps to duplicate it?

**Comment by fr1sty** [comment]
I'm sorry for the confusion.
The List(s) Import Manager is still not doing anything, the numbers displayed there are incorrect, they don't change when adding movies/tvshows to my watchlist on Trakt. And when pressing sync it only gives me a message 'Updating library from trakt lists' and nothing further. I thought before i would get more messages, about completing even though it didnt do anything. 

The numbers in the List(s) Import Manager only change if i do the 'Force Trakt Sync to local database' option.

I was under the impression that the 'Trakt service sync interval(in minutes)' would sync my watchlist to Umbrella. Which after that would sync into my library. But i can see that there are no files created in my umbrella movies/tvshows folders. So there is nothing to import for the Kodi library sync.

**Comment by umbrellaplug** [comment]
Okay, I'm going to write out a lot more extra logging. I do see from the log you posted earlier the new line I added to give me the exception shows it's failing in the "add" section of the library import code. Without explaining the ins and outs of the code here, a file that isn't going to be added because it's missing information should never even make it to this point in the code. That's the other error you're seeing about super.info(). Items missing information are removed from the list before they are sent to the code to add the files. This really shouldn't prevent trakt sync so I'm wondering what's going on with that. 

Watchlist and Collection are actually different on syncing. I know, I know. Confusing.

Collection was something that was in the addon before I took over and is syncing your entire "collected" shows/movies from trakt every however many hours you have set for the interval in the setting for it in trakt section of settings. That's why it's on a different interval. I've loaded a fresh kodi load 10.03 on my pi and I'm going to attempt to duplicate your issue here. 

On the Endurance show giving you errors, I'm not sure how long it takes for trakt to realize tmdb has been updated and send the id with the list, but you'll need to clear your meta to get those new ids on the import once it does update.

Watchlist, List you've made, and Liked List will show in import mananger window and use the Trakt Sync interval (usually 15 minutes) to update the items on the list. They import every 1-24 hours depending on what you have set for trakt list import in settings> library

Collection should be updated every 15 minutes with the trakt sync as well, but uses the interval from the Settings> Trakt> Auto Trakt Collections Syncing

I'll post another reply here when I have the updated test version with more logging posted.

**Comment by umbrellaplug** [comment]
6.5.90 is on the test repo now. It should resolve your issues for you. There was a control removed from LibreElec for internet state that I was checking in the code to run the Trakt sync. That code has been in there since the venom days, so there's no telling how long that hasn't worked. Install new version and reboot and you should see trakt lists start to update within whatever time you have set for trakt sync cycle. Let me know if this resolves the issue for you.

**Comment by fr1sty** [comment]
Good to hear you found something. I already installed version 6.5.90 yesterday. To get the changes do I need to uninstall and reinstall this version? Because the repository is not notifying me of a new version.

**Comment by umbrellaplug** [comment]
Yeah there were two versions of 6.5.90 I pushed yesterday. Sorry didn't know you had already grabbed the first one. You'd want to just reinstall then reboot kodi and should be good.

**Comment by umbrellaplug** [comment]
Quick update, I'd leave this alone till I you hear back from me. I got the internetstate stuff fixed and running on my side, but I added a movie to my collection and left it overnight on the pi and it was not added. I need to track down what is happening. I will push a version .91 and let you know once I've had a chance to look over this.

**Comment by umbrellaplug** [comment]
Just pushed .91. Hopefully that resolved it for you. I've been watching mine this morning it's been updating so far. I have logging setup for each time it checks.

**Comment by fr1sty** [comment]
My movies watchlist disappeared from the List(s) Import Manager and got some error for you :) Seems like maybe your Trakt account is hardcoded?
`[2022-11-24 20:04:18] [ Umbrella: DEBUG ]: From func name: resources.lib.modules.trakt.sync_user_lists() Line # :1493
                       msg : Trakt User Lists Sync Update...(local db latest "lists_updatedat" = 72000, trakt api latest "lists_updatedat" = 1669021729)
[2022-11-24 20:04:18] [ Umbrella: DEBUG ]: From func name: resources.lib.modules.trakt.sync_liked_lists() Line # :1517
                       msg : Trakt Liked Lists Sync Update...(local db latest "liked_at" = 72000, trakt api latest "liked_at" = 1669021729)
[2022-11-24 20:04:19] [ Umbrella: DEBUG ]: From func name: resources.lib.modules.trakt.sync_trending_lists() Line # :1668
                       msg : Trakt Trending Lists Sync Update...(local db latest "trendinglist_at" = 1669056361, cache expiry = 1669230259)
[2022-11-24 20:04:30] [ Umbrella: INFO ]: (CarlsonPictureShow) has marked their list private in Trakt(Trending Lists) and is now causing you errors. Skipping this list
[2022-11-24 20:04:30] [ Umbrella: INFO ]: (CarlsonPictureShow) has marked their list private in Trakt(Trending Lists) and is now causing you errors. Skipping this list
[2022-11-24 20:04:34] [ Umbrella: WARNING ]: getTrakt() (404:NOT FOUND): URL=(https://api.trakt.tv/users/2borno2b/lists/2826229/items/movie,show):
[2022-11-24 20:06:19] [ Umbrella: DEBUG ]: From func name: resources.lib.menus.tvshows.super_info() Line # :1080
                       msg : ### TV Show Super Function: ids={imdb: tt8291284, tmdb: 95403, tvdb: 349187}
[2022-11-24 20:06:19] [ Umbrella: DEBUG ]: From func name: resources.lib.menus.tvshows.super_info() Line # :1080
                       msg : ### TV Show Super Function: ids={imdb: tt11280740, tmdb: 95396, tvdb: 371980}
[2022-11-24 20:19:47] [ Umbrella: DEBUG ]: From func name: resources.lib.modules.trakt.sync_user_lists() Line # :1493
                       msg : Trakt User Lists Sync Update...(local db latest "lists_updatedat" = 72000, trakt api latest "lists_updatedat" = 1669021729)
[2022-11-24 20:19:48] [ Umbrella: DEBUG ]: From func name: resources.lib.modules.trakt.sync_liked_lists() Line # :1517
                       msg : Trakt Liked Lists Sync Update...(local db latest "liked_at" = 72000, trakt api latest "liked_at" = 1669021729)
`

**Comment by umbrellaplug** [comment]
Can you get the full log? This snippet isn't helpful.

**Comment by fr1sty** [comment]
This is all there is after
`[2022-11-24 20:04:09] [ Umbrella: INFO ]: ######   UMBRELLA SERVICE ENTERING KEEP ALIVE   #####
[2022-11-24 20:04:13] [ Umbrella: INFO ]: #################### STARTING TRAKT SCHEDULING ################
[2022-11-24 20:04:13] [ Umbrella: INFO ]: #################### SCHEDULED TIME FRAME 1 HOURS ###############
`

**Comment by umbrellaplug** [comment]
That's not the full log. There's a lot of logging statements to do with syncing your actual list that are before this. It makes it extremely difficult to troubleshoot without the full log. I'm going to go ahead and close this issue out.

**Comment by fr1sty** [comment]
Ok, I'm sorry I could not help you with the right logs in time. Thanks for your effort!

**Comment by adamosborne83** [comment]
> Ok, I'm sorry I could not help you with the right logs in time. Thanks for your effort!

Kodi Log uploaded would allow you to post the whole log.
Https://kodi.paste.tv/

**Comment by fr1sty** [comment]
[Link to my umbrella.log](https://paste.kodi.tv/ojamowucev.kodi) 
Don't know if its right but I'm seeing username 2borno2b in the logs.

**Comment by umbrellaplug** [comment]
The username your seeing for lists is from the trakt trending lists. Looks like something failed in your watchlist sync.

**Comment by umbrellaplug** [comment]
I just looked over the code and found the problem. If you restart kodi right now without doing anything it's likely your items will be added to library. 

**For some reason the trakt sync collection code was written to only run 1 time.**

You did have an issue with internetstate. That's a very good catch all libreelec devices will have that issue due to that setting being removed. That has been fixed in this version. Sorry to close your issue out on you earlier, but it actually is fixed I wanted to re-open the issue and elaborate.

InternetState setting missing caused Trakt Sync not to run unless ran using Forced. I have changed that and have been monitoring it all day. It appears to be running as expected. here's a snippet of my log so you can see what I'm looking for to determine if the sync is happening.

`2022-11-24 15:01:31.980 T:1099     INFO <general>: [ plugin.video.umbrella ]  Trakt Service-Sync (abort requested not detected. checking has connection)
2022-11-24 15:01:31.984 T:1099     INFO <general>: [ plugin.video.umbrella ] Trakt Service Sync InternetState: False 
2022-11-24 15:01:31.984 T:1099     INFO <general>: [ plugin.video.umbrella ] Trakt Service Sync HasNetwork: True `

This should be in the log every 15 minutes if you have the trakt sync set to that. I had mine set to 5 minutes. I don't recommend this I was just being impatient.

All this being said, Lists you import occur based on the setting you have set for library interval in settings, with the exclusion of trakt collection. I excluded it because I thought it was a bad idea to import the entire collection. To resolve this for you and work the way I think anyone would expect it to work I'm going to move the collection option into import lists with watchlists, your lists, and liked lists. I will be removing the trakt sync collection section from the settings in trakt. This will put all the lists in one place and should avoid any confusion like this going forward. I have no idea what was intended originally when setting that up to only occur once, but I'm sure Venom had a reason for it. We both assumed it was a repeating task.

In Summary:
Trakt Syncs every 15 minutes default.
Trakt Collection sync only runs once. Twice if set to run on start then again at whatever time interval.
Library Service Runs every 6 Hours by default and can be changed using a slider and does repeat.

Trakt Collection will be moved into list available to import and lists available to import will be changed to be pulled fresh each time instead of caching.

I will reply here with an updated version once I make these changes. Moving this to enhancement from bug fix.

**Comment by umbrellaplug** [comment]
Just pushed latest version 6.5.92. It should fix all your issues. If you don't mind can you please try it out and let me know. All settings for importing lists are now found under library settings.

**Comment by umbrellaplug** [comment]
No response in two days.

**Comment by fr1sty** [comment]
I see I was just too late too comment before closing. My family needed some attention and I wanted to test something before putting it down here. Somehow it seems that if I add a new item to my Trakt list on the Trakt website, it disappears after a few minutes. Same for when i remove an item, it will automatically reappear in my Trakt list. 
Going to remove the Trakt connection from Kodi now, to see if this has anything to do with that.

**Comment by umbrellaplug** [comment]
Yeah. No worries. I understand. I figured you were upset I closed it the other day and just gave up responding. 

The current version on test repo should get you all fixed up. You may want to clear database on each list, but really you can just manually import using the option below that now and it won't even check. Manual import just brings the lists in no matter what was previously there. 

I've warmed up a bit to removing items as well. I really hate to get into removing from library but it seems like that might be a good option for these lists to keep library from growing out of control. It took me several minutes to remove your watchlist from my library yesterday from testing.

**Comment by fr1sty** [comment]
First of all, just want to say that I really appreciate the job you're doing here!
Secondly I solved the case of the disappearing items. I have a IMDB script which syncs my IMDB watchlist to Trakt, and I discovered that it removes items which are not in my IMDB list. I never use the Trakt site, but now for testing purposes I did. 
I have to configure Trakt on my kodi again. I will come back to you and let you know if I succeed this time :)

**Comment by fr1sty** [comment]
I added some movies and tv shows to my list. After one sync I can see it added one tv show in the umbrella directory. But it skipped the rest of the shows and movies. The syncs after that didn't add anything.

**Comment by umbrellaplug** [comment]
When you say it skipped the rest of the shows and movies, what do you mean?

What list did you add these to (watchlist, collection)? Are the ones you added showing in the list on My TV Shows or My Movies?

What do you mean after one sync it added one show in the umbrella directory? A a strm file or what? Sorry, just trying to understand what's happening.

**Comment by fr1sty** [comment]
I added movies and shows to my Trakt watchlist. 
I can see the umbrella sync added one of the shows, because i can see it created the directory and strm files. (But now i can see it was created at 14:15 earlier today, so this doesn't help)

The rest of the shows and movies that i added to Trakt watchlist were not created. 
I checked My Tv Shows and My Movies, but they are not there. Even the show that synced earlier today is not there. 
I also checked the Lists Import manager and the numbers there did not change, the show that has been added did not change the number. 

I try to explain it as best as possible, but I understand it is confusing.

**Comment by umbrellaplug** [comment]
Okay, can you pull a recent log? I can look and see if any indications of what may be happening there.

**Comment by fr1sty** [comment]
[Umbrella log](https://paste.kodi.tv/vogarayawa.kodi) 
The 'The provided authorization grant is invalid' warning is because i revoked the permission for a few minutes.

**Comment by umbrellaplug** [comment]
Perfect, something killed your trakt watchlist sync at `[2022-11-26 15:12:02]`. I will see if I can add more logging here and try to duplicate your watchlist and see if I can get the same thing. I'll let you know what I find.

**Comment by umbrellaplug** [comment]
Can you tell if it's your show or movie watchlist it stopped on? I'd like to try to import the same list here.

**Comment by fr1sty** [comment]
I will test it syncing my shows watchlist only.

**Comment by fr1sty** [comment]
I selected only my shows link to sync, and did a reboot. I also want to mention that the lists import manager is still showing old numbers. 
I don't see any files being created in the umbrella shows dir. 
[Log file](https://paste.kodi.tv/vepihedeso.kodi)

**Comment by umbrellaplug** [comment]
Okay, yeah something has caused your trakt watchlist to stop syncing. I planned on getting Trakt VIP today so I can import others lists. That's why I was asking which list it was. I'll add some more logging in for the watchlist stuff and reduce the logging for collection and try it out.

**Comment by umbrellaplug** [comment]
Do you mind trying the latest version on the test repo. 6.5.93

I did a fresh install of Kodi, loaded this version, added my trakt, added my library folders, then added your watchlist as my watchlist and set my watchlist to import automatically. Within the hour I had 21 shows in my library. I've added some new logging into the watchlist sync, I believe you may have some corruption in your traktsync.db. One thing I can say is I develop using windows and you're running on a linux. I'm loading my pi up with these versions to make sure there isn't something missing for a dependency. Please load the latest version, see if it syncs watchlist and grab the full log and send it back afterwards. 

Thanks again, sorry to still be working on this. Latest version has the added surprise of source select and auto play separated for tv and movies though.

**Comment by fr1sty** [comment]
I'm still happy that you choose to spend time in trying to fix this. 
I installed the update, revoked and authorized trakt, rebooted. I noticed no differences, also no change in numbers of shows in the Trakt lists. Umbrella still says that there are 18 shows while I have 21 on Trakt. 
Deleted the traktSync.db file, rebooted, file got created again. Still 18 shows and no difference in behaviour. Are there any other files I can delete manually?
[Here is my logfile](https://paste.kodi.tv/ovuqoloqiw.kodi)

**Comment by fr1sty** [comment]
Well it seems that everything went fine now! All my movies and series have been added automatically! :)

**Comment by umbrellaplug** [comment]
Found a couple more small things. One was lists being cleared incorrectly from the import manager window. If you can please try the latest version on the test repo. You will need to manually update. I have changed the test versions to use a 3 digit number instead of 2 for version checks so the version numbers don't get bumped up too much with test versions, that's why it will require a manual update. 

You will want 6.5.942 or higher. 

If you have any issues please let me know. I'll continue to monitor this for a bit before pushing this version to the main repo.

**Comment by umbrellaplug** [comment]
Everything still going ok for you? My pi seems to be syncing and updating as expected now. I think this one may be resolved.

**Comment by fr1sty** [comment]
It seems that the sync for my shows broke after removing and adding some shows on trakt.

**Comment by umbrellaplug** [comment]
Can you upload a log?

**Comment by fr1sty** [comment]
I can see an error at the hourly sync. The log is full with the 5 minute sync, going to adjust that to 15 mins again. 
I'm also going to remove and add some movies to my movie list. 
[log](https://paste.kodi.tv/uqudovunud.kodi)

**Comment by umbrellaplug** [comment]
I'm going to close this issue as I believe the Trakt Lists Import Manager is working. If you have some type of other trakt sync issue or something feel free to open another issue, but this one needs to be closed. I appreciate the heads up for the internetstate connection issue and have made several changes to improve imports for everyone based on your feedback.

**Comment by fr1sty** [comment]
The shows import stopped working. I can see that there are no files being created for the latest added shows. 
When doing a clean import, i can see it tries to import all the shows but it says that the items already exist in the library. While they are not there. 

Have some errors in the log file. Does deleting traktSync.db help? Movie import is still working.

[Log file](https://paste.kodi.tv/evovujewaw.kodi)

**Comment by umbrellaplug** [comment]
Turn off include unknown air dates.

**Comment by fr1sty** [comment]
Did not change anything, still the log is full with those errors. 
After clearing list database umbrella and running import its still saying that the new shows exist. I have my kodi database in an mariadb and i can see that the new shows are not there. 
Deleted traktSync.db and library.db but both did not give any results.

**Comment by umbrellaplug** [comment]
I don't know, you're commenting on an issue that's literally for something else and closed and almost 2 months old with logs that have once again been trimmed so can't actually see the real cause of the error. this is ridiculous. I'm deleting this issue so you'll stop commenting on a closed issue.
